source/extensions/quic_listeners/quiche/envoy_quic_connection.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#pragma GCC diagnostic push
+// QUICHE allows unused parameters.
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+// QUICHE uses offsetof().
+#pragma GCC diagnostic ignored "-Winvalid-offsetof"
+
+#include "quiche/quic/core/quic_connection.h"
+
+#pragma GCC diagnostic pop
+
+#include <memory>
+
+#include "common/common/logger.h"
+#include "envoy/network/connection.h"
+#include "envoy/network/listener.h"
+#include "server/connection_handler_impl.h"
+
+namespace Envoy {
+namespace Quic {
+
+// Derived for network filter chain, stats and QoS.
+class EnvoyQuicConnection : public quic::QuicConnection,
+                            protected Logger::Loggable<Logger::Id::connection> {
+public:
+  EnvoyQuicConnection(const quic::QuicConnectionId& server_connection_id,
+                      quic::QuicSocketAddress initial_peer_address,
+                      quic::QuicConnectionHelperInterface* helper,
+                      quic::QuicAlarmFactory* alarm_factory, quic::QuicPacketWriter* writer,
+                      bool owns_writer, quic::Perspective perspective,
+                      const quic::ParsedQuicVersionVector& supported_versions,
+                      Network::ListenerConfig& listener_config,
+                      Server::ListenerStats& listener_stats);
+
+  ~EnvoyQuicConnection() override = default;
+
+  // Called by EnvoyQuicSession::setConnectionStats().
+  void setConnectionStats(const Network::Connection::ConnectionStats& stats) {
```

Comment:
setConnectionStats -> SetConnectionStats ?

Doing a general pass for non-overridden functions which have googleCasing is worthwhile actually